base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1259 @@
+      !call MAPL_ConfigSetAttribute(config, dateline, 'DATELINE:')
+
+      call ESMF_VMGetCurrent(vm, rc=status)
+      _VERIFY(status)
+      call ESMF_VMGet(vm, PETcount=nPet, rc=status)
+      _VERIFY(status)
+
+      nx_guess = nint(sqrt(real(nPet)))
+      do nx = nx_guess,1,-1
+         ny=nPet/nx
+         if (nx*ny==nPet) then
+            call MAPL_ConfigSetAttribute(config, nx, 'NX:')
+            call MAPL_ConfigSetAttribute(config, ny, 'NY:')
+            exit
+         end if
+      enddo
+
+      call this%initialize(config, rc=status)
+      _VERIFY(status)
+
+
+   end subroutine initialize_from_esmf_distGrid
+
+   function decomps_are_equal(this,a) result(equal)
+      class (SwathGridFactory), intent(in) :: this
+      class (AbstractGridFactory), intent(in) :: a
+      logical :: equal
+
+      select type (a)
+         class default
+         equal = .false.
+         return
+      class is (SwathGridFactory)
+         equal = .true.
+
+
+         equal = size(a%ims)==size(this%ims) .and. size(a%jms)==size(this%jms)
+         if (.not. equal) return
+
+         ! same decomposition
+         equal = all(a%ims == this%ims) .and. all(a%jms == this%jms)
+         if (.not. equal) return
+
+      end select
+
+   end function decomps_are_equal
+
+
+   function physical_params_are_equal(this, a) result(equal)
+      class (SwathGridFactory), intent(in) :: this
+      class (AbstractGridFactory), intent(in) :: a
+      logical :: equal
+
+      select type (a)
+         class default
+         equal = .false.
+         return
+      class is (SwathGridFactory)
+         equal = .true.
+
+         equal = (a%im_world == this%im_world) .and. (a%jm_world == this%jm_world)
+         if (.not. equal) return
+
+         !-- prepare to delete it
+         !equal = &
+         !     & all(a%lon_centers == this%lon_centers) .and. &
+         !     & all(a%lat_centers == this%lat_centers)
+      end select
+
+   end function physical_params_are_equal
+
+   logical function equals(a, b)
+      class (SwathGridFactory), intent(in) :: a
+      class (AbstractGridFactory), intent(in) :: b
+
+      select type (b)
+         class default
+         equals = .false.
+         return
+      class is (SwathGridFactory)
+         equals = .true.
+
+         equals = (a%lm == b%lm)
+         if (.not. equals) return
+
+         equals = a%decomps_are_equal(b)
+         if (.not. equals) return
+
+         equals = a%physical_params_are_equal(b)
+         if (.not. equals) return
+
+      end select
+
+   end function equals
+
+
+   function generate_grid_name(this) result(name)
+      character(len=:), allocatable :: name
+      class (SwathGridFactory), intent(in) :: this
+
+      character(len=4) :: im_string, jm_string
+
+      write(im_string,'(i4.4)') this%im_world
+      write(jm_string,'(i4.4)') this%jm_world
+
+      !      name = this%dateline // im_string // 'x' // jm_string
+      name = im_string // 'x' // jm_string
```

Comment:
@bena-nasa Is there a problem that this grid name may overlap our names for LatLon?   Maybe we need something like "swath" as a prefix or suffix?   And maybe even the filename that provided the geolocations?